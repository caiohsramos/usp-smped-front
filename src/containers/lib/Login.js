import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container } from './Container';
import { LoginController } from '../../controllers';
import { LoginForm, LoginLogo, LoginFooter } from '../../components';
import { setToken, clearToken } from '../../actions/TokenActions';
import { bindActionCreators } from 'redux'

class Login extends Container {
    constructor(props) {
        super(props);

        this.state = {
            EMAIL: '',
            PASSWORD: '',
            LOGINERROR: '',
        };

        const toController = {
            callback: this.callback,
            getState: this.getState,
            getProps: this.getProps,
            router: props.router,
            setToken: props.setToken,
            clearToken: props.clearToken
        };

        this.controller = new LoginController(toController);
    }

    render() {
        const { handleChangeAction, handleSubmitAction } = this.controller;
        const { EMAIL, PASSWORD, LOGINERROR } = this.state;
        return (
        <div className='login'>
                <LoginLogo />
                <LoginForm
                    email={EMAIL}
                    password={PASSWORD}
                    loginError={LOGINERROR}
                    changeAction={handleChangeAction}
                    submitAction={handleSubmitAction}
                />
                <LoginFooter/>
            </div>
        );
    }
}

const mstp = state => {
    return {};
};

const mdtp = dispatch => 
    bindActionCreators({setToken, clearToken}, dispatch);

export default connect(mstp, mdtp)(Login);
