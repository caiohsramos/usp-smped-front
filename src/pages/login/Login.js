import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container } from '../Container';
import { LoginController } from './LoginController';
import { LoginForm } from './components/LoginForm';
import { setToken } from '../../actions/TokenActions';
import { bindActionCreators } from 'redux';
import { Logo } from '../../common';

const LoginLogo = props => {
	const { } = props;

	return (
        <div className='login-logo'>
            <Logo/>
            <div></div>
        </div>
	);
};

const LoginFooter = props => {
	const { } = props;

	return (
        <div className='login-foot'>
            LabXP IME-USP Â© 2018
        </div>
	);
};

class Login extends Container {
	constructor(props) {
		super(props);
		this.state = {
			EMAIL: '',
			PASSWORD: '',
			LOGINERROR: '',
		};

		const toController = {
			callback: this.callback,
			getState: this.getState,
			getProps: this.getProps,
			router: props.router,
			setToken: props.setToken,
		};

		this.controller = new LoginController(toController);
	}

	render() {
		const { handleChangeAction, handleSubmitAction } = this.controller;
		const { EMAIL, PASSWORD, LOGINERROR } = this.state;
		return (
            <div className='login'>
                <LoginLogo />
                <LoginForm
                    email={EMAIL}
                    password={PASSWORD}
                    loginError={LOGINERROR}
                    changeAction={handleChangeAction}
                    submitAction={handleSubmitAction}
                />
                <LoginFooter/>
            </div>
		);
	}
}

const mstp = state => {
	return {};
};

const mdtp = dispatch => 
    bindActionCreators({setToken}, dispatch);

export default connect(mstp, mdtp)(Login);


