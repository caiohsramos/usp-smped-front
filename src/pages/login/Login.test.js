import React from 'react';
import Login from './Login';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import reducer from '../../reducers';

const store = createStore(reducer);

it('renders correctly', () => {
    const tree = renderer
        .create(<Login store={store} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});