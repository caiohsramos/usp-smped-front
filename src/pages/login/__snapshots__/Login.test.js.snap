// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="login"
>
  <div
    className="login-logo"
  >
    <div
      className="logo"
    >
      <img
        alt="login-logo"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/90/Bras%C3%A3o_da_cidade_de_S%C3%A3o_Paulo.svg/200px-Bras%C3%A3o_da_cidade_de_S%C3%A3o_Paulo.svg.png"
      />
    </div>
    <div />
  </div>
  <form
    className="login-form"
  >
    <div
      className="text-input"
      id="text-input-EMAIL"
    >
      <input
        className="input"
        id="EMAIL"
        onChange={[Function]}
        placeholder="Endereço de email"
        type="text"
        value=""
      />
    </div>
    <div
      className="text-input"
      id="text-input-PASSWORD"
    >
      <input
        className="input"
        id="PASSWORD"
        onChange={[Function]}
        placeholder="Senha"
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-lg btn-primary btn-block"
      id="submit"
      onClick={[Function]}
      type="submit"
    >
       
      Entrar
       
    </button>
  </form>
  <div
    className="login-foot"
  >
    LabXP IME-USP © 2018
  </div>
</div>
`;
